pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -188,7 +188,9 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 	// always update the elasticsearch state bits
 	d.ReconcileState.UpdateElasticsearchState(*resourcesState, observedState())
 
-	if err := d.verifySupportsExistingPods(resourcesState.CurrentPods); err != nil {
+	_, permissive := d.ES.Annotations[esv1.DisableDowngradeValidationAnnotation]
```

Comment:
I wonder if we should try to detect when the user set `false`, and expect the validation to be enabled (similar to what we do for `eck.k8s.elastic.co/managed`)